calliope/backend/pyomo/constraints/costs.py

```diff
@@ -227,13 +234,16 @@ def cost_var_constraint_rule(backend_model, cost, loc_tech, timestep):
     else:
         cost_prod = 0
 
-    if loc_tech_is_in(backend_model, loc_tech, "loc_techs_supply_plus") and cost_om_con:
+    if loc_tech_is_in(backend_model, loc_tech, "loc_techs_supply_plus") and is_valid(
+        cost_om_con
+    ):
         cost_con = cost_om_con * weight * backend_model.resource_con[loc_tech, timestep]
-    elif loc_tech_is_in(backend_model, loc_tech, "loc_techs_supply") and cost_om_con:
+    elif loc_tech_is_in(backend_model, loc_tech, "loc_techs_supply") and is_valid(
+        cost_om_con
+    ):
         energy_eff = get_param(backend_model, "energy_eff", (loc_tech, timestep))
-        if (
-            po.value(energy_eff) > 0
-        ):  # in case energy_eff is zero, to avoid an infinite value
+        # in case energy_eff is zero, to avoid an infinite value
+        if po.value(energy_eff) > 0:
```

Comment:
I think you missed an `is_valid` check, line 254?

A clear separation between valid and invalid is probably easier to reason, so a separate `if`-clause for the validity check might be cleaner:

    if is_valid(..): # or not invalid(..)
        cost_con = 0
    else:
        if loc_tech_is_in(..):
            ...
        elif loc_tech_is_in(..):
            ...